Userland/Libraries/LibCore/Stream.h

```diff
@@ -113,8 +113,8 @@ class Stream {
     ErrorOr<T> read_trivial_value()
     {
         alignas(T) u8 buffer[sizeof(T)] = {};
-        TRY(read_entire_buffer(buffer));
-        return *bit_cast<T>(buffer);
+        TRY(read_entire_buffer({ &buffer, sizeof(buffer) }));
```

Comment:
Alright I spent more time than I care to admit (:smiling_face_with_tear:) trying to figure out why this line needed to change, because it *should* work.

Turns out that while we define this constructor on `AK::Detail::Span<T>`:
```c++
    template<size_t size>
    ALWAYS_INLINE constexpr Span(T (&values)[size])
        : m_values(values)
        , m_size(size)
    {
    }
```

We do *not* define such a constructor on the `AK::Detail::Span<u8>` nor the `AK::Detail::Span<u8 const>` specializations. Seems to me like those specializations should be declared against the non-Detail `Span` definition, but I bet that'll unfold a can of compiler worms.

In the meantime, we could define such a constructor on those specializations. Not gonna ask you to fix that here though, just wanted to document what was going on, because the original line you wrote here *should* been fine.